integration/spark2/src/main/scala/org/apache/spark/sql/hive/CarbonSessionState.scala

```diff
@@ -81,34 +81,38 @@ class CarbonSessionCatalog(
   override def lookupRelation(name: TableIdentifier,
       alias: Option[String]): LogicalPlan = {
     val rtnRelation = super.lookupRelation(name, alias)
-    rtnRelation match {
-      case SubqueryAlias(_,
-          LogicalRelation(carbonDatasourceHadoopRelation: CarbonDatasourceHadoopRelation, _, _),
-          _) =>
-        refreshRelationFromCache(name, alias, carbonDatasourceHadoopRelation)
+    var toRefreshRelation = false
+    rtnRelation foreach {
```

Comment:
Better add `SubqueryAlias(_, LogicalRelation(carbonDatasourceHadoopRelation: CarbonDatasourceHadoopRelation, _, _)` condition as well. There might be some scenarios it may come, so better add this as well.